src/components/form-elements/fieldset/Fieldset.tsx

```diff
@@ -44,60 +44,11 @@ interface FieldsetProps extends HTMLProps<HTMLFieldSetElement> {
   disableErrorLine?: boolean;
 }
 
-const FieldSet = ({ className, disableErrorLine, fieldsetRef, ...rest }: FieldsetProps) => {
-  const [registeredComponents, setRegisteredComponents] = useState<string[]>([]);
-  const [erroredComponents, setErroredComponents] = useState<string[]>([]);
-
-  const passError = (componentId: string, error: boolean): void => {
-    const existingError = erroredComponents.includes(componentId);
-    if (existingError && !error) {
-      setErroredComponents(erroredComponents.filter((id) => id !== componentId));
-      return;
-    }
-    if (!existingError && error) {
-      setErroredComponents([...erroredComponents, componentId]);
-    }
-  };
-
-  const registerComponent = (componentId: string, deregister = false): void => {
-    let newComponents = [...registeredComponents];
-    if (deregister) {
-      newComponents = newComponents.filter((id) => id !== componentId);
-    } else if (!registeredComponents.includes(componentId)) {
-      newComponents = [...newComponents, componentId];
-    }
-    setRegisteredComponents(newComponents);
-  };
-
-  const contextValue: IFieldsetContext = useMemo(() => {
-    return {
-      isFieldset: true,
-      registerComponent: registerComponent,
-      passError: passError,
-    };
-  }, [registerComponent, passError]);
-
-  const containsFormElements = registeredComponents.length > 0;
-  const containsError = erroredComponents.length > 0;
-
+const FieldSet = ({ className, fieldsetRef, disableErrorLine, ...rest }: FieldsetProps) => {
   return (
-    <FieldsetContext.Provider value={contextValue}>
-      {containsFormElements ? (
-        <div
-          className={classNames('nhsuk-form-group', {
-            'nhsuk-form-group--error': disableErrorLine ? false : containsError,
-          })}
-        >
-          <fieldset
-            className={classNames('nhsuk-fieldset', className)}
-            ref={fieldsetRef}
-            {...rest}
-          />
-        </div>
-      ) : (
-        <fieldset className={classNames('nhsuk-fieldset', className)} ref={fieldsetRef} {...rest} />
-      )}
-    </FieldsetContext.Provider>
+    <FormGroup enableErrorLine={!disableErrorLine}>
```

Comment:
I realise we're doing this for backwards compatibility, but the internal property being the opposite of the public option feels confusing. One to think about in the future?

Reply:
Yeah the main reason this was done was so that consumers of the new `FormGroup` component (when using outside of `Fieldset`) had to actively opt-in to enable the error line decoration rather than it being enabled by default. Outside of the context of using within a `Fieldset` it seems like a specific case to have that error line rather than it being the default one